### PR TITLE
Add Docker /v2/_catalog endpoint

### PR DESCRIPTION
**Pull request by KN4CK3R**
This PR adds the `/v2/_catalog` endpoint.

Can be tested with `docker-ls`:
grafik

related to #20111

### CODE REVIEW

**Review by silverwind — APPROVED** [review]
Rubberstamp LGTM

File: models/migrations/v220.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	container_module "code.gitea.io/gitea/modules/packages/container"
+
+	"xorm.io/xorm"
+	"xorm.io/xorm/schemas"
+)
+
+func addContainerRepositoryProperty(x *xorm.Engine) error {
+	switch x.Dialect().URI().DBType {
+	case schemas.SQLITE:
+	case schemas.ORACLE:
```
Comment (lunny) [inline models/migrations/v220.go]: `ORACLE` is not supported yet.

Comment (KN4CK3R) [reply models/migrations/v220.go]: Other migrations use `ORACLE` too (110, 210).

Comment (lunny) [reply models/migrations/v220.go]: Yeah, just realize them. But I think we should also delete them. What do you think about that?

Comment (KN4CK3R) [reply models/migrations/v220.go]: Sure, I don't need it. 😄

Comment (KN4CK3R) [inline models/migrations/v220.go]:
```suggestion
```

**Review by lunny — APPROVED** [review]

**Comment by wxiaoguang** [comment]
bot

**Comment by 6543** [comment]
-> #20556

**Pull request merged**